wip: add regenerating blocks and tnt

Simon-Laux opened the pull request:
- add yarn lockfile
- make it start without webxdc.js
- add self healing bricks
- bricks now only give score once they are destroyed
- add tnt

Simon-Laux committed:
add yarn lockfile

Simon-Laux committed:
make it start without webxdc.js

Simon-Laux committed:
add self healing bricks

Simon-Laux committed:
bricks now only give score once they are destroyed

Simon-Laux committed:
add tnt

adbenitez commented on js/levels.js:
```diff
@@ -2,16 +2,25 @@ window.levels = {
-        [2, 0, 2, 0, 2, 0, 2],
+        [2, 0, 2, 5, 2, 0, 2],
        ],
+    flags_1: [
```
maybe better to convert every level in a object with board and flags property (or check if it is object or in legacy array format)

Simon-Laux replied:
yeah also thought that, but as you don't use typescript nor jsdoc, such refactorings need more attention to not introduce bugs, so I focused on user facing stuff.

adbenitez commented on images/tnt_bricks.png:
these TNT images don't fit with the sci-fi, space, futuristic theme of the game, we need to replace them in the future, what is the meaning of a TNT in an indestructible/white block?

adbenitez replied:
ex. something like:
![nuclear1](https://github.com/DeltaZen/arkanoid.xdc/assets/24558636/e707e4b9-4d4a-4318-92a7-6ecc0feccc00)


taken from https://opengameart.org/content/basic-nuclear-signs

adbenitez replied:
I will probably try something fancy in the future, like the brick is broken in the middle reveling the sign inside like a broken cracked wall or so :D

adbenitez replied:
icons we could use for the self-generating blocks and +1 life blocks:

https://opengameart.org/content/mana-health-and-energy-icons

ex. the green one would fit in the self-generating blocks

adbenitez replied:
also found this dynamite that is pixel art and will fit better with the existing graphics: https://opengameart.org/content/dynamite-0

Simon-Laux replied:
> what is the meaning of a TNT in an indestructible/white block?

It was late, I didn't payed attention, maybe you can troll people with that. we could also make it explode if a ant nearby explodes if we want, or we could revert the design

Simon-Laux replied:
> icons we could use for the self-generating blocks

my idea for now was that they are healthy (not cracked), but icon can also make sense, I just didn't want to look for one at the time.

adbenitez commented:
looking really promising! this PR will open the possibility to also add blocks that give you +1 life, etc. in the future :star_struck:

adbenitez commented:
to avoid having a lot of images and increasing app size a lot, would be nice to come with some dynamic generation of the special blocks, like just drawing the icon on top of the blocks or having the blocks with some broken crack in the middle with transparency and drawing the icon then the block mask on top

Simon-Laux committed:
explode other tnt,
regardless wether brick was destroyed or not.
white brick's tnt explodes, but white brick stays

Simon-Laux committed:
fix explosion radius

Simon-Laux committed:
add sfx and zzfx

Simon-Laux committed:
add explosion graphic

though location is still wrong in game, but right when I set it manually

Simon-Laux committed:
fix explosion position

Simon-Laux committed:
add explosion animation

Simon-Laux committed:
fix tnt destroying white blocks

Simon-Laux committed:
fix: undefined block are invisible indestructible blocks

Simon-Laux committed:
delayed explosions

fix double explosions

adbenitez commented:
pssst! @Simon-Laux  could you create a branch now from within this repo then copy the files overriding them or doing the changes there somehow and override/force-push this PR or create a new one so I can also collaborate directly to this branch/PR? 

other option is that I send you the files and you do the commit then replacing them

Simon-Laux committed:
start adding drops

adbenitez commented on js/sfx.js:
```diff
@@ -0,0 +1,11 @@
+
+export const playExplosionSFX = ()=> {
+    if (Math.random() >=.5){
+        zzfx(...[1.01,,690,.02,.3,.55,4,3.03,.8,,,,,1.8,,.2,,.5,.12,.11])
```
it will be good to move all our sounds to this type, so we can remove the howler dependency and not to have two dependencies for sounds

Simon-Laux committed:
add sounds

Simon-Laux committed:
new dynamite image, that adb gave me

Simon-Laux commented:
Superseded by #7
